eureka_ml_insights/data_utils/transform.py

```diff
@@ -302,3 +303,40 @@ def transform(self, df: pd.DataFrame, encoding="cl100k_base") -> pd.DataFrame:
             token_count_column = f"{column}_token_count"
             df[token_count_column] = token_count
         return df
+
+@dataclass
+class MajorityVoteTransform:
+    """Applies the majority vote transformation to the 'model_output' column per 'ID'."""
+    
+    # No need for additional parameters as the logic is fixed for majority vote.
+    
+    def transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Transforms the dataframe by calculating the majority vote of 'model_output' per 'ID'.
+        If the 'model_output' is NaN, it will be handled appropriately.
```

Comment:
This is vague. Let's explicitly mention that NaNs are dropped in calculating majority.